packages/flutter/test/material/paginated_data_table_test.dart

```diff
@@ -1043,4 +1044,83 @@ void main() {
     await tester.pumpWidget(buildFrame(overflowBar));
     expect(headerX, tester.getTopLeft(find.byType(ElevatedButton)).dx);
   });
+
+  testWidgets('PaginatedDataTable can be scrolled using ScrollController', (WidgetTester tester) async {
+    final TestDataSource source = TestDataSource();
+    final ScrollController scrollController = ScrollController();
+
+    Widget buildTable(TestDataSource source) {
+      return Align(
+        alignment: Alignment.topLeft,
+        child: SizedBox(
+          width: 100,
+          child: PaginatedDataTable(
+            controller: scrollController,
+            header: const Text('Test table'),
+            source: source,
+            rowsPerPage: 2,
+            columns: const <DataColumn>[
+              DataColumn(
+                label: Text('Name'),
+                tooltip: 'Name',
+              ),
+              DataColumn(
+                label: Text('Calories'),
+                tooltip: 'Calories',
+                numeric: true,
+              ),
+              DataColumn(
+                label: Text('Generation'),
+                tooltip: 'Generation',
+              ),
+            ],
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(MaterialApp(
+      home: buildTable(source),
+    ));
+
+    expect(scrollController.offset, 0.0);
+
+    scrollController.jumpTo(50.0);
+    await tester.pumpAndSettle();
+    expect(scrollController.offset, 50.0);
+  });
+
+  testWidgets('PaginatedDataTable uses PrimaryScrollController when primary ', (WidgetTester tester) async {
+    final ScrollController primaryScrollController = ScrollController();
+    final TestDataSource source = TestDataSource();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: PrimaryScrollController(
+          controller: primaryScrollController,
+          child: PaginatedDataTable(
+            primary: true,
+            header: const Text('Test table'),
+            source: source,
+            rowsPerPage: 2,
+            columns: const <DataColumn>[
+              DataColumn(label: Text('Name')),
+              DataColumn(label: Text('Calories'), numeric: true),
+              DataColumn(label: Text('Generation')),
+            ],
+          ),
+        ),
+      )
+    );
+
+    // PaginatedDataTable has two Scrollable Widgets, First one is for DataTable
+    // while second one is meant for Footer.
```

Comment:
is this line required?

Reply:
What do you mean?

Reply:
There are two scrollable in the PaginatedDataTable, so should this comment be there for explanation?

Reply:
Sure! It never hurst to have extra comments explaining things. Future contributors will appreciate it. :)